agent/provider/src/payments/payment_checker.rs

```diff
@@ -0,0 +1,499 @@
+use actix::prelude::*;
+use anyhow::anyhow;
+use chrono::{DateTime, Duration, Utc};
+use std::collections::HashMap;
+
+use ya_utils_actix::{
+    actix_signal::{SignalSlot, Subscribe},
+    actix_signal_handler,
+};
+
+pub trait Deadlineable: Clone + Ord + Sized + Send + Sync + Unpin + 'static {}
+
+/// Will be sent when deadline elapsed.
+#[derive(Message, Clone, Debug)]
+#[rtype(result = "()")]
+pub struct DeadlineElapsed {
+    pub agreement_id: String,
+    pub deadline: DateTime<Utc>,
```

Comment:
here the `deadline` is proper name, as it's describing specific point of the time